src/picongpu/include/particles/ionization/byField/ADK/AlgorithmADK.hpp

```diff
@@ -82,8 +85,21 @@ namespace ionization
                 float_X dBase = float_X(4.0) * util::cube(protonNumber) / (eInAU * util::quad(nEff)) ;
                 float_X dFromADK = math::pow(dBase,nEff);
 
+                float_X polarizationFactor;
+                /* set up the polarization factor */
+                if(T_linPol)
```

Comment:
@PrometheusPi  Your suggestion is not possible because like @n01r said the factor is only known at runtime.

A possible solution would be to add a functor for the factor calculation but it is a little bit complicated because of the needed input values.

My suggestion: (small design change)

```C++
/* ionization rate */
float_X rateADK = polarizationFactor \
                * eInAU * util::square(dFromADK) / (float_X(8.0) * pi * protonNumber) \
                * math::exp(float_X(-2.0) * util::cube(protonNumber) / (float_X(3.0) * util::cube(nEff) * eInAU));

if(T_linPol)
{
    /* factor from averaging over one laser cycle with LINEAR polarization */
    const float_X polarizationFactor = math::sqrt(float_X(3.0) * util::cube(nEff) * eInAU / (pi * util::cube(protonNumber)));
    rateADK *= polarizationFactor;
}
``` 
Then the code is better readable.